contracts/verifier/BoltzDeployVerifier.sol

```diff
@@ -48,25 +49,33 @@ contract BoltzDeployVerifier is
 
         destinationContractValidation(relayRequest.request.to);
 
-        if (relayRequest.request.tokenContract != address(0)) {
-            require(
-                tokens[relayRequest.request.tokenContract],
-                "Token contract not allowed"
-            );
+        if (relayRequest.request.tokenAmount > 0) {
+            if (relayRequest.request.tokenContract != address(0)) {
+                require(
+                    tokens[relayRequest.request.tokenContract],
+                    "Token contract not allowed"
+                );
 
-            require(
-                relayRequest.request.tokenAmount <=
-                    IERC20(relayRequest.request.tokenContract).balanceOf(
-                        contractAddr
-                    ),
-                "Token balance too low"
-            );
-        } else {
-            require(
-                relayRequest.request.tokenAmount <=
-                    address(contractAddr).balance,
-                "Native balance too low"
-            );
+                require(
+                    relayRequest.request.tokenAmount <=
+                        IERC20(relayRequest.request.tokenContract).balanceOf(
+                            contractAddr
+                        ),
+                    "Token balance too low"
+                );
+            } else {
+                uint256 amount = BoltzUtils.validateClaim(
+                    relayRequest.request.data,
+                    relayRequest.request.to,
+                    contractAddr
+                );
+
+                require(
+                    relayRequest.request.tokenAmount <=
+                        address(contractAddr).balance + amount,
```

Comment:
yes, we didn't want to refactor all the variables because this would have been a breaking change, so where possible we renamed it, while here it's still `tokenAmount` even if it refers to RBTC